ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/ArtifactRegistryIntegrationForm.tsx

```diff
@@ -32,34 +34,47 @@ export type ArtifactRegistryIntegrationFormValues = {
 
 export const validationSchema = yup.object().shape({
     config: yup.object().shape({
-        name: yup.string().required('An integration name is required'),
+        name: yup.string().trim().required('An integration name is required'),
         categories: yup
             .array()
-            .of(yup.string().oneOf(['REGISTRY']))
+            .of(yup.string().trim().oneOf(['REGISTRY']))
             .min(1, 'Must have at least one type selected')
             .required('A category is required'),
         google: yup.object().shape({
-            endpoint: yup.string().required('An endpoint is required'),
-            project: yup.string().required('A project is required'),
+            endpoint: yup.string().trim().required('An endpoint is required'),
+            project: yup.string().trim().required('A project is required'),
+            wifEnabled: yup.boolean(),
             serviceAccount: yup
                 .string()
-                .test(
-                    'serviceAccount-test',
-                    'A service account key is required',
-                    (value, context: yup.TestContext) => {
-                        const requirePasswordField =
-                            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-                            // @ts-ignore
-                            context?.from[2]?.value?.updatePassword || false;
-
-                        if (!requirePasswordField) {
-                            return true;
-                        }
-
-                        const trimmedValue = value?.trim();
-                        return !!trimmedValue;
-                    }
-                ),
+                .trim()
+                .when('wifEnabled', {
+                    is: false,
+                    then: (serviceAccountSchema) =>
+                        serviceAccountSchema
+                            .required('A service account key is required')
+                            .test(
+                                'isValidJson',
+                                'Service account key must be valid JSON',
+                                (value, context: yup.TestContext) => {
+                                    const isRequired =
```

Comment:
nit: I think leaving the name of this variable as `requirePasswordField` is more clear than the vague `isRequired`